src/backend/plaf.rs

```diff
@@ -0,0 +1,285 @@
+        self.advice_index = advice_index;
+        self.fixed_index = fixed_index;
+
+        if !self.circuit.polys.is_empty() {
+            let mut counter = Counter::new(); 
+            for c_poly in &mut self.circuit.polys.iter() {
+                let plaf_poly = pPoly {
+                    name: c_poly.annotation.clone(),
+                    exp: self.convert_plaf_poly(&c_poly.expr, &mut counter),
+                };
+                plaf.polys.push(plaf_poly);
+            }
+        }
+
+        for lookup in self.circuit.lookups.iter() {
+            let mut counter = Counter::new();
+
+            let v1 = lookup
+                .exprs
+                .clone()
+                .into_iter()
+                .map(|(e1, _)| self.convert_plaf_poly(&e1, &mut counter)) // counter needs to be supplied but not really used here
+                .collect::<Vec<pExpr<PlonkVar>>>();
+
+            let v2 = lookup
+                .exprs
+                .clone()
+                .into_iter()
+                .map(|(_, e2)| self.convert_plaf_poly(&e2, &mut counter))  // counter needs to be supplied but not really used here
+                .collect::<Vec<pExpr<PlonkVar>>>();
+            
+            let plaf_lookup = pLookup {
+                name: lookup.annotation.clone(),
+                exps: (v1, v2),
+            };
+
+            plaf.lookups.push(plaf_lookup);
+        }
+
+        plaf
+    }
+
+    fn convert_and_push_plaf_column(
+        &self, 
+        column: &cColumn, 
+        plaf: &mut Plaf, 
+        // the three Option fields need to be all Some or all None; 
+        // not the best practice but this function is only used interally
+        c_column_id_to_p_column_index: Option<&mut HashMap<u32, usize>>, 
+        advice_index: Option<&mut Counter>,
+        fixed_index: Option<&mut Counter>,
+    ) {
+        match column.ctype {
+            cAdvice => {
+                let plaf_witness = ColumnWitness::new( // advice is called witness in plaf
+                    column.annotation.clone(),
+                    column.phase,
+                );
+                // will panic if c_column_id_to_p_column_index is Some but advice_index is None
+                self.add_id_index_mapping(column, c_column_id_to_p_column_index, advice_index.unwrap()); 
+                plaf.columns.witness.push(plaf_witness); 
+            }
+            cFixed => {
+                let plaf_fixed = ColumnFixed::new(column.annotation.clone());
+                self.add_id_index_mapping(column, c_column_id_to_p_column_index, fixed_index.unwrap());
+                plaf.columns.fixed.push(plaf_fixed);
+            }
+            Halo2Advice => { // ??? QUESTION: should terminate with error but only phase is missing from chiquito imported halo2 advice, so I defaulted phase to 0. is this good?
+                let plaf_witness = ColumnWitness::new(
+                    column.annotation.clone(),
```

Comment:
When the user imports a column from Halo2 expects it to be the same column to share info, but you are creating a new column in plaf that is not the same as the Halo2 column. The only solution is to emit an error

Reply:
Fixed by emitting an error. Could you explain what you mean by "not the same as the Halo2 column"? My logic was that Plaf fixed column only requires a name string, which the imported halo2 fixed can supply, so I think we can port it from halo2 to plaf, but I might be missing something. Similar for advice column as well.

Reply:
The import of columns is to share the column with another halo2 circuit. So the creates plaf column is not same as that halo2 column. Like in the bytecode circuit you share the column with the keccak circuit or push data circuit to lookup. If you create this new column it does not have the same witness or is actually the same.

Reply:
Ah this makes much more sense from a higher level. Thanks for the explanation!

Reply:
Also I used panic! for all error handling to follow the same practice in other Chiquito files. This is not the end game "proper" error handling but I'm assuming this is fine for now :)